store/sqlstore/channel_store.go

```diff
@@ -3446,8 +3560,68 @@ func (s SqlChannelStore) getSearchGroupChannelsQuery(userId, term string, isPost
 	return query, args
 }
 
-// TODO: rewrite in squirrel (https://github.com/mattermost/mattermost-server/issues/19335)
-func (s SqlChannelStore) SearchGroupChannels(userId, term string) (model.ChannelList, error) {
+func (s SqlChannelStore) squirrelGetSearchGroupChannelsQuery(userId, term string, isPostgreSQL bool) sq.SelectBuilder {
+	var baseLikeTerm string
+	var selectQuery sq.SelectBuilder
+
+	terms := strings.Split(strings.ToLower(strings.Trim(term, " ")), " ")
+
+	having := sq.And{}
+
+	if isPostgreSQL {
+		baseLikeTerm = "ARRAY_TO_STRING(ARRAY_AGG(u.Username), ', ') LIKE ?"
+		cc := sq.Select("c.Id").
+			From("Channels c").
+			Join("ChannelMembers cm ON c.Id=cm.ChannelId").
+			Join("Users u on u.Id = cm.UserId").
+			Where(sq.And{sq.Eq{"c.Type": "G"}, sq.Eq{"u.id": userId}}).
+			GroupBy("c.Id")
+
+		for _, term := range terms {
+			term = sanitizeSearchTerm(term, "\\")
+			having = append(having, sq.Expr(baseLikeTerm, "%"+term+"%"))
+		}
+
+		subq := sq.Select("cc.id").
+			FromSelect(cc, "cc").
+			Join("ChannelMembers cm On cc.Id = cm.ChannelId").
+			Join("Users u On u.Id = cm.UserId").
+			GroupBy("cc.Id").
+			Having(having).
+			Limit(model.ChannelSearchDefaultLimit)
+
+		selectQuery = sq.Select("*").
+			From("Channels").
+			Where(sq.Expr("Id IN (?)", subq))
+
+	} else {
+		baseLikeTerm = "GROUP_CONCAT(u.Username SEPARATOR ', ') LIKE ?"
+
+		for _, term := range terms {
+			term = sanitizeSearchTerm(term, "\\")
+			having = append(having, sq.Expr(baseLikeTerm, "%"+term+"%"))
+		}
+
+		cc := sq.Select("c.*").
+			From("Channels c").
+			Join("ChannelMembers cm ON c.Id=cm.ChannelId").
+			Join("Users u on u.Id = cm.UserId").
+			Where(sq.And{sq.Eq{"c.Type": "G"}, sq.Eq{"u.Id": userId}}).
+			GroupBy("c.Id")
+
+		selectQuery = sq.Select("cc.*").
+			FromSelect(cc, "cc").
+			Join("ChannelMembers cm on cc.Id = cm.ChannelId").
+			Join("Users u on u.Id = cm.UserId").
+			GroupBy("cc.Id").
+			Having(having).
+			Limit(model.ChannelSearchDefaultLimit)
+	}
+
+	return selectQuery
+}
+
+func (s SqlChannelStore) SqlSearchGroupChannels(userId, term string) (model.ChannelList, error) {
```

Comment:
Please update the original methods themselves. There's no point keeping unused code lying around.

Reply:
Thought I removed them all. Thanks for catching that.